dynamic-proxy/src/proxy.rs

```diff
@@ -0,0 +1,153 @@
+use crate::{
+    body::{simple_empty_body, to_simple_body, SimpleBody},
+    request::should_upgrade,
+    upgrade::{split_request, split_response, UpgradeHandler},
+};
+use http::StatusCode;
+use hyper::{Request, Response};
+use hyper_util::{
+    client::legacy::{connect::HttpConnector, Client},
+    rt::TokioExecutor,
+};
+use std::{convert::Infallible, time::Duration};
+
+const DEFAULT_TIMEOUT: Duration = Duration::from_secs(10);
+
+/// A client for proxying HTTP requests to an upstream server.
+pub struct ProxyClient {
+    client: Client<HttpConnector, SimpleBody>,
+    timeout: Duration,
+}
+
+impl Clone for ProxyClient {
```

Comment:
seems it can! This was probably initially a generic `struct` (which derived types sometimes screw up), and when I removed the type parameter I didn't notice that the `Clone` could be simplified.